utils/env_test.go

```diff
@@ -0,0 +1,91 @@
+package utils
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestMergeDicts(t *testing.T) {
+	MergeDicts(nil, nil)
+	d1 := map[string]interface{}{
+		"t": "1",
+	}
+	d2 := map[string]interface{}{
+		"t":  "2",
+		"t2": "1",
+	}
+	expect := func(m1, m2 map[string]interface{}, v ...string) {
+		d := MergeDicts(m1, m2)
+		for i := 0; i < len(v)/2; i++ {
+			if fmt.Sprintf("%v", d[v[i*2]]) != v[i*2+1] {
+				t.Errorf("err %v, expected %s", d, v)
+			}
+		}
+	}
+	expect(nil, d1, "t", "1")
+	expect(d1, d1, "t", "1")
+	expect(d1, nil, "t", "1")
+	expect(d2, d1, "t", "1", "t2", "1")
+	expect(d1, d2, "t", "2", "t2", "1")
+	expect(nil, d2, "t", "2", "t2", "1")
+}
+
+func TestMergeDictsRecursive(t *testing.T) {
+	d0 := map[string]interface{}{
+		"a": "1",
+	}
+	d1 := map[string]interface{}{
+		"a": "11",
+		"b": "2",
+	}
+	r0 := map[string]interface{}{
+		"r":  d0,
+		"r1": "1",
+	}
+	r1 := map[string]interface{}{
+		"r":  d1,
+		"r2": "2",
+	}
+	r3 := map[string]interface{}{
+		"r":  "00",
+		"r2": "0",
+	}
+
+	expect := func(m1, m2 map[string]interface{}, v string) {
+		d := MergeDicts(m1, m2)
+		if fmt.Sprintf("%v", d) != v {
+			t.Errorf("err %v, expected %s", d, v)
+		}
+	}
```

Comment:
The maps comparison should be done with `cmp.Equal` from `"github.com/google/go-cmp/cmp"`
Comparing the rendered strings does not guarantee types equality.

Reply:
Does it matter?, since MergeMaps cannot return anything except `map[string]interface{}` and all these values & results are manufactured test cases.

Reply:
This is Go's testing best practices, and it does matter, 
`map[string]interface{}{"key": int(1)}` is different from `map[string]interface{}{"key": uint(1)}` but their string representation is the same